### PR TITLE
Init Project

### PR DESCRIPTION
**Pull request by umairah97**
ready

### CODE REVIEW

**Comment by umairah97** [comment]
ready

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,62 @@
+source 'https://rubygems.org'
+git_source(:github) { |repo| "https://github.com/#{repo}.git" }
+
+ruby '2.7.0'
+
+# Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
+gem 'rails', '~> 5.2.3'
+# Use sqlite3 as the database for Active Record
+gem 'sqlite3'
```
Comment (ngocvt-0484) [inline Gemfile:9]: use gem mysql2 instead of sqlite3
after that, you bundle install and restart server

File: Gemfile [hunk 2]
```diff
@@ -0,0 +1,62 @@
+source 'https://rubygems.org'
+git_source(:github) { |repo| "https://github.com/#{repo}.git" }
+
+ruby '2.7.0'
+
+# Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
+gem 'rails', '~> 5.2.3'
+# Use sqlite3 as the database for Active Record
```
Comment (ngocvt-0484) [inline Gemfile:8]: remove all comment

File: Gemfile [hunk 3]
```diff
@@ -0,0 +1,62 @@
+source 'https://rubygems.org'
+git_source(:github) { |repo| "https://github.com/#{repo}.git" }
+
+ruby '2.7.0'
+
+# Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
+gem 'rails', '~> 5.2.3'
```
Comment (ngocvt-0484) [inline Gemfile:7]: you should use double quote in rails file

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>SampleApp</title>
+    <%= csrf_meta_tags %>
+    <%= csp_meta_tag %>
+
+    <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload' %>
```
Comment (ngocvt-0484) [inline app/views/layouts/application.html.erb:8]: use double quote

File: config/database.yml [hunk 5]
```diff
@@ -0,0 +1,25 @@
+# SQLite version 3.x
+#   gem install sqlite3
+#
+#   Ensure the SQLite 3 gem is defined in your Gemfile
+#   gem 'sqlite3'
+#
```
Comment (ngocvt-0484) [inline config/database.yml:6]: this file will not be up to GIt, please add it to gitignore
`cp database.yml database.yml.example
git rm --cached config/database.yml`
and in .gitignore add this line to ignore it
`/config/database.yml`

Comment (ngocvt-0484) [reply config/database.yml:6]: after that, you 
```
git add .
git commit --amend --no-edit
git push your_branch -f
```

File: config/credentials.yml.enc [hunk 6]
```diff
@@ -0,0 +1 @@
+XfpgsqT1n51D3k308B1oZy6R4ifB5SWaeT5IS9P7kAnPGEiKlI24ofmhXdkcpGW35LgfNyC7bmSLm9wfT0dNXHdCAKQ4c7uHWiU0cjJgxr/2uI6Dzr15T/jecRIsCyc3zupo4ZZBSPZksdnAwp0b52Cd8wHmlx8MJbl2DMPqlTtjeDi0RRhdl38VynuIinxt3KFRb8wrkThuBwVqvyDgS+78Z1fg+vo/vRXYKSXIw/k0kVVSSbRjTvLGwdlBt8PPOsok+AVrdbSFjKiuBmCQqHxa8w06bHiSPWIk5QMIrUOVuK3uF0z+wtmLGnQ/tGHgiBggeYBDUyqwU1BZ3kEKX22oNSJZx53mRhGq0lavx27SZlyZp4LfACnnYxKuwxRu/EZQ/GN4eQiGuGxFi4O73L4C7Oo7aXweJL7t--O0t7DNIYoQLkrl1H--jwqNhGL4aegKutjeCJQDOg==
```
Comment (ngocvt-0484) [inline config/credentials.yml.enc:1]: similar with database.yml

File: config/locales/en.yml [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+# the default I18n backend:
+#
+# true, false, on, off, yes, no
+#
+# Instead, surround them with single quotes.
+#
+# en:
+#   'true': 'foo'
+#
+# To learn more, please read the Rails Internationalization guide
```
Comment (ngocvt-0484) [inline config/locales/en.yml:29]: remove all comment

File: test/test_helper.rb [hunk 8]
```diff
@@ -0,0 +1,10 @@
+ENV['RAILS_ENV'] ||= 'test'
+require_relative '../config/environment'
+require 'rails/test_help'
+
+class ActiveSupport::TestCase
```
Comment (ngocvt-0484) [inline test/test_helper.rb:5]: dont need do test
you can add it to gitignore

File: Gemfile (hunk 1 above)
Comment (umairah97) [reply Gemfile:9]: when i execute gem rails, it says commandLineError. what should i do.

Comment (umairah97) [reply Gemfile:9]: even when i execute gem install mysql2 it says FilePermissionError

Comment (umairah97) [reply Gemfile:9]: ah i understand

**Comment by umairah97** [comment]
ready

File: Gemfile [hunk 9]
```diff
@@ -0,0 +1,50 @@
+source 'https://rubygems.org'
+git_source(:github) { |repo| "https://github.com/#{repo}.git" }
+
+ruby '2.7.0'
+
+
+gem "rails", "~> 5.2.3"
+
+gem 'mysql2'
+bundle install
```
Comment (ngocvt-0484) [inline Gemfile:10]: ??
why you add this code?
please remove it

File: Gemfile [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+gem 'puma', '~> 3.11'
```
Comment (ngocvt-0484) [inline Gemfile:11]: use double quote

File: Gemfile [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
+gem 'sass-rails', '~> 5.0'
+
+gem 'uglifier', '>= 1.3.0'
+
+
```
Comment (ngocvt-0484) [inline Gemfile:17]: remove empty line

File: config/database.yml [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+# re-generated from your development database when you run "rake".
+# Do not set this db to the same as development or production.
+test:
+  <<: *default
+  database: db/test.sqlite3
+
+production:
+  <<: *default
+  database: db/production.sqlite3
+cp database.yml database.yml.example
```
Comment (ngocvt-0484) [inline config/database.yml:26]: you shoud run it in terminal, not put here

File: config/locales/en.yml [hunk 13]
```diff
@@ -0,0 +1,4 @@
+
+
```
Comment (ngocvt-0484) [inline config/locales/en.yml:2]: remove empty line

**Comment by ngocvt-0484** [comment]
please change 10 commits to 1 commit
you can use `git rebase -i HEAD~10`

File: config/database.yml (hunk 12 above)
Comment (umairah97) [reply config/database.yml:26]: oh,im sorry, i misunderstood. is it for every file or just the one that has been corrected?

Comment (umairah97) [reply config/database.yml:26]: i understand. i will run in terminal. should i just leave it there or delete it?

**Comment by umairah97** [comment]
ready

**Comment by thaodtd-0665** [comment]
ready

**Comment by thaodtd-0665** [comment]
ready

**Comment by thaodtd-0665** [comment]
commented

**Comment by thaodtd-0665** [comment]
ready

File: .ruby-version [hunk 14]
```diff
@@ -0,0 +1 @@
+ruby-2.7.0
```
Comment (saobv-2223) [inline .ruby-version:1]: Add an empty line after line 2, please

File: Gemfile [hunk 15]
```diff
@@ -0,0 +1,33 @@
+source 'https://rubygems.org'
+git_source(:github) { |repo| "https://github.com/#{repo}.git" }
+ruby '2.7.0'
```
Comment (saobv-2223) [inline Gemfile:3]: using " instead of ' in ruby files. please

File: Gemfile [hunk 16]
```diff
@@ -0,0 +1,33 @@
...
+gem "coffee-rails", "~> 4.2"
+gem "turbolinks", "~> 5"
+gem "jbuilder", "~> 2.5"
+gem "bootsnap", ">= 1.1.0", require: false
+group :development, :test do
+  gem "byebug", platforms: [:mri, :mingw, :x64_mingw]
+end
+group :development do
+ gem "web-console", ">= 3.3.0"
+  gem 'listen', '>= 3.0.5', '< 3.2'
```
Comment (saobv-2223) [inline Gemfile:18]: using "

File: app/mailers/application_mailer.rb [hunk 17]
```diff
@@ -0,0 +1,4 @@
+class ApplicationMailer < ActionMailer::Base
+  default from: 'from@example.com'
+  layout 'mailer'
```
Comment (saobv-2223) [inline app/mailers/application_mailer.rb:3]: using "

File: app/views/layouts/application.html.erb [hunk 18]
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>SampleApp</title>
+    <%= csrf_meta_tags %>
+    <%= csp_meta_tag %>
+
+    <%= stylesheet_link_tag    "application", media: "all", "data-turbolinks-track": "reload" %>
+    <%= javascript_include_tag 'application', 'data-turbolinks-track': 'reload' %>
```
Comment (saobv-2223) [inline app/views/layouts/application.html.erb:9]:
```suggestion
    <%= stylesheet_link_tag "application", media: "all", "data-turbolinks-track": "reload" %>
    <%= javascript_include_tag "application", "data-turbolinks-track": "reload" %>
```

File: test/application_system_test_case.rb [hunk 19]
```diff
@@ -0,0 +1,5 @@
+require "test_helper"
+
```
Comment (saobv-2223) [inline test/application_system_test_case.rb:2]: open `.gitignore` and add line `test`
run `git rm --cached -r test`

File: .ruby-version (hunk 14 above)
Comment (umairah97) [reply .ruby-version]: line 2 is going to be empty too right? does that mean line 2 and line 3 are empty?

File: Gemfile (hunk 15 above)
Comment (umairah97) [reply Gemfile]: done

File: Gemfile (hunk 16 above)
Comment (umairah97) [reply Gemfile]: done

**Pull request closed without merging**